### ISSUE TITLE
_TestConnection() function

### ISSUE THREAD

**Issue by resuni** [WHMCS/sample-provisioning-module#8]
I can't get the "_TestConnection()" function to work as described here: https://github.com/WHMCS/sample-provisioning-module/blob/master/modules/servers/provisioningmodule/provisioningmodule.php#L381

I return an array just like the example shows, but if I press the "Test Connection" button in the server settings, a loading wheel next to the button spins indefinitely. I know my code is executing fine, because I can use the "logModuleCall()" function to log expected values to the module debug log right before I return the array. 

I also notice that "_TestConnection()" is not listed as a supported function in the documentation: https://developers.whmcs.com/provisioning-modules/supported-functions/

It seems like something about this function isn't documented right.

**Comment by resuni** [comment]
For anyone else that has this spinning wheel problem, I was able to fix it by switching from PHP 5.6 to PHP 7.2. 

I'll close this issue, but I still think this function needs better documentation. For example, it looks like it's only possible to display a custom message when the test fails. Can a custom message be displayed when the test succeeds instead of just displaying "SUCCESS!"?

**Comment by leandrogbreve** [comment]
Hello community,

I'm getting this same error using PHP 7.4. When I click the button to test, the loading wheel bar is shown spinning forever.

I'm not too expert in developing, but I believe my code should work. I'm working overwritting the code found in the sample-provisioning-module. From what I understand I need to validate the connection inside de try block which, I'm doing with **$params** and the **ss2_auth_password** function to return true or false (I got this function from another module that has start, finish and reboot functions that work properly).

My code is as below: 

```
function neoxenserver_TestConnection(array $params)
{
	$host = $params["serverip"];
	$port = 22;
	$user = $params["serverusername"];
	$pass = $params["serverpassword"];
	$usessl = $params["serversecure"];
	$connection = ssh2_connect("$host", $port);
    
    try {
        // Call the service's connection test function.
	$success = ssh2_auth_password($connection, $user , $pass);
        $errorMsg = '';
    } catch (Exception $e) {
        // Record the error in WHMCS's module log.
        logModuleCall(
            'provisioningmodule',
            __FUNCTION__,
            $params,
            $e->getMessage(),
            $e->getTraceAsString()
        );

        $success = false;
        $errorMsg = $e->getMessage();
    }

    return array(
        'success' => $success,
        'error' => $errorMsg,
    );
}
```
My browser console shows: 

**[WHMCS] Error: error** pointing to **scripts.min.js?v=13898b:26**

I'm suspect that, logically thinking, I'm not sure if the input fields are available in $params, and if not, how can I develop the function in PHP by using the values inputed in the page?

I hope someone can give orientation. Thanks in advance.
